### PR TITLE
ASoC: SOF: add trace event support

### PR DESCRIPTION
**Pull request by bardliao**
Add trace event to show the trace log from firmware.

---
We call the trace event function when host_offset is updated since it indicates firmware flush the dma buffer.

### CODE REVIEW

File: include/trace/events/intel-sof.h [hunk 1]
```diff
@@ -0,0 +1,138 @@
+/* SPDX-License-Identifier: GPL-2.0 */
+#undef TRACE_SYSTEM
+#define TRACE_SYSTEM intel-sof
```
Comment (lgirdwood) [inline include/trace/events/intel-sof.h:3]: rename to  sof.h

File: include/trace/events/intel-sof.h [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+/*
+ * The TRACE_SYSTEM_VAR defaults to TRACE_SYSTEM, but must be a
+ * legitimate C variable. It is not exported to user space.
+ */
+#undef TRACE_SYSTEM_VAR
+#define TRACE_SYSTEM_VAR intel_sof
```
Comment (lgirdwood) [inline include/trace/events/intel-sof.h:10]: rename to sof

File: include/trace/events/intel-sof.h [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+
+#if !defined(_TRACE_INTEL_SOF_H) || defined(TRACE_HEADER_MULTI_READ)
```
Comment (lgirdwood) [inline include/trace/events/intel-sof.h:12]: in general remove "intel" from this file and use sof instead.

File: include/trace/events/intel-sof.h [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+#define _TRACE_INTEL_SOF_H
+
+#include <linux/types.h>
+#include <linux/ktime.h>
+#include <linux/tracepoint.h>
+
+/* should be the same as rmbox */
+#define TRACE_CLASS_IRQ		(1 << 24)
```
Comment (lgirdwood) [inline include/trace/events/intel-sof.h:20]: We cant decode this here, it must be decoded in userspace. i.e. we should emit a timestamp and then ASCII HEX output of the data (unless it's common data among all trace). Please also see the lates tarce updates on master branch for ABI update.

File: sound/soc/sof/trace.c [hunk 5]
```diff
@@ -247,6 +251,8 @@ int snd_sof_trace_update_pos(struct snd_sof_dev *sdev,
 			     struct sof_ipc_dma_trace_posn *posn)
 {
 	if (sdev->dtrace_is_enabled && sdev->host_offset != posn->host_offset) {
+		snd_sof_trace_read_buf(sdev,
```
Comment (lgirdwood) [inline sound/soc/sof/trace.c:254]: I would also expect code here to remove the debugFS trace file.

File: include/trace/events/intel-sof.h (hunk 4 above)
Comment (bardliao) [reply include/trace/events/intel-sof.h:20]: @lgirdwood So the output data will look like "14ff3e 19259f", right? The first element is the timestamp and the second one is the data. Also we need an user space tool to parse these output data? Could you show me the commit id, please. I will look at it.

File: sound/soc/sof/trace.c (hunk 5 above)
Comment (bardliao) [reply sound/soc/sof/trace.c:254]: @lgirdwood Should we remove it after all trace_event function is ready? I.e. include the user space tool. Or can we let trace event and debugFS exist simultaneously for a while to make sure we didn't miss anything in trace event?

File: include/trace/events/intel-sof.h (hunk 4 above)
Comment (ranj063) [reply include/trace/events/intel-sof.h:20]: @bardliao rmbox does the parsing. its part of the soft repo

File: sound/soc/sof/trace.c (hunk 5 above)
Comment (lgirdwood) [reply sound/soc/sof/trace.c:254]: lets remove in the same PR, but have a new PR for soft that has the updates to decode the trace data.

**Comment by plbossart** [comment]
is this PR still relevant? It's been stale for 27 days, either it's needed or it needs to be closed.

**Comment by bardliao** [comment]
@plbossart I think we can close it ATM.

**Pull request closed without merging**